File: src/main/java/org/vladg/io/scanner/ProjectScanner.java
package org.vladg.io.scanner;

import org.vladg.ast.ASTCreator;
import org.vladg.jafax.utils.filefinder.FileFinder;

import java.nio.file.Path;

public class ProjectScanner {

    public static void beginScan(Path path) {
        var files = FileFinder.findFiles(path);
        var javaFiles = files.javaFiles.toArray(new String[0]);
        var jarFiles = files.jarFiles.toArray(new String[0]);
        ASTCreator.createAst(javaFiles, jarFiles);
    }

}

File: src/main/java/org/vladg/io/scanner/ProjectScanner.kt
package org.vladg.io.scanner

import org.vladg.ast.ASTCreator
import org.vladg.jafax.utils.filefinder.FileFinder
import org.vladg.logger
import java.nio.file.Path

object ProjectScanner {

    private val logger = logger();

    fun beginScan(path: Path) {
        logger.info("Scanning for files...")
        val files = FileFinder.findFiles(path)
        val javaFiles = files.javaFiles.toTypedArray()
        val jarFiles = files.jarFiles.toTypedArray()
        ASTCreator.createAst(javaFiles, jarFiles)
    }

}